ocs_ci/ocs/dr/dr_workload.py

```diff
@@ -136,48 +136,177 @@ def __init__(self, **kwargs):
+            ]["name"]
+            drpc_yaml_data["spec"]["preferredCluster"] = cluster
+            drpc_yaml_data["spec"]["drPolicyRef"]["name"] = self.dr_policy_name
+            templating.dump_data_to_temp_yaml(drpc_yaml_data, self.drpc_yaml_file)
+
+            # TODO
+            # drpc_yaml_file needs to be committed back to the repo
+            # because ACM would refetch from repo directly
+
+            # load channel.yaml
+            channel_yaml_data = templating.load_yaml(self.channel_yaml_file)
+            channel_yaml_data["metadata"]["name"] = helpers.create_unique_resource_name(
+                resource_type="channel", resource_description="ramen-gitops"
+            )
+            channel_yaml_data["spec"]["pathname"] = self.workload_repo_url
+            templating.dump_data_to_temp_yaml(channel_yaml_data, self.channel_yaml_file)
 
-        # load channel.yaml
-        channel_yaml_data = templating.load_yaml(self.channel_yaml_file)
-        channel_yaml_data["spec"]["pathname"] = self.workload_repo_url
-        templating.dump_data_to_temp_yaml(channel_yaml_data, self.channel_yaml_file)
+            # load git-repo namespace.yaml
+            git_ns_yaml_data = templating.load_yaml(self.git_repo_namespace_yaml_file)
+            git_ns_yaml_data["metadata"]["name"] = helpers.create_unique_resource_name(
+                resource_type="namespace", resource_description="ramen-busybox"
+            )
+            templating.dump_data_to_temp_yaml(
+                git_ns_yaml_data, self.git_repo_namespace_yaml_file
+            )
 
-        # Create the resources on Hub cluster
-        config.switch_acm_ctx()
-        run_cmd(f"oc create -k {self.workload_subscription_dir}")
-        run_cmd(f"oc create -k {self.workload_subscription_dir}/{self.workload_name}")
+            # load subscription.yaml
+            subscription_yaml_data = templating.load_yaml(self.subscription_yaml_file)
+            subscription_yaml_data["metadata"][
+                "name"
+            ] = helpers.create_unique_resource_name(
+                resource_type="subscription", resource_description="busybox"
+            )
+            subscription_yaml_data["spec"]["channel"] = (
+                git_ns_yaml_data["metadata"]["name"]
+                + "/"
+                + channel_yaml_data["metadata"]["name"]
+            )
+            subscription_yaml_data["spec"]["placement"]["placementRef"][
+                "name"
+            ] = placementrule_yaml_data["metadata"]["name"]
+            templating.dump_data_to_temp_yaml(
+                subscription_yaml_data, self.subscription_yaml_file
+            )
 
-        self.verify_workload_deployment()
+            # load app yaml
+            app_yaml_data = templating.load_yaml(self.app_yaml_file)
+            app_yaml_data["metadata"]["name"] = helpers.create_unique_resource_name(
+                resource_type="app", resource_description="busybox"
+            )
```

Comment:
why we are create unique resources?

Reply:
we use same repository and same path to create apps, so it deploys say "busybox-workloads1" on both the clusters with same namespaces, since we cannot have apps on same namespaces on both primary and secondary, hence bringing up this uniqueness